packages/outstatic/src/pages/edit-document/index.tsx

```diff
@@ -70,33 +74,130 @@ export default function EditDocument({ collection }: EditDocumentProps) {
 
-      const commitInput = createCommitInput({
+      const capi = createCommitApi({
+        message: oldSlug
+          ? `chore: Updates ${newSlug} formerly ${oldSlug}`
+          : `chore: Updates/Creates ${newSlug}`,
         owner,
-        slug: newSlug,
-        oldSlug,
-        content,
-        oid,
-        files,
-        repoSlug,
-        repoBranch,
-        contentPath,
-        monorepoPath,
-        collection
+        oid: oid ?? '',
+        name: repoSlug,
+        branch: repoBranch
       })
 
-      await createCommit({ variables: commitInput })
+      if (oldSlug) {
+        capi.removeFile(
+          `${
+            monorepoPath ? monorepoPath + '/' : ''
+          }${contentPath}/${collection}/${oldSlug}.md`
+        )
+      }
+
+      if (files.length > 0) {
+        files.forEach(({ filename, blob, type, content: fileContents }) => {
+          // check if blob is still in the document before adding file to the commit
+          if (blob && content.search(blob) !== -1) {
+            const randString = window
+              .btoa(Math.random().toString())
+              .substring(10, 6)
+            const newFilename = filename
+              .toLowerCase()
+              .replace(/[^a-zA-Z0-9-_\.]/g, '-')
+              .replace(/(\.[^\.]*)?$/, `-${randString}$1`)
+
+            const filePath = (() => {
+              switch (type) {
+                case 'images':
+                  return IMAGES_PATH
+                default:
+                  assertUnreachable(type)
+              }
+            })()
+
+            capi.replaceFile(
+              `${
+                monorepoPath ? monorepoPath + '/' : ''
+              }public/${filePath}${newFilename}`,
+              fileContents,
+              false
+            )
+
+            // replace blob in content with path
+            content = content.replace(blob, `/${filePath}${newFilename}`)
+          }
+        })
+      }
+
+      capi.replaceFile(fullContentPath, content)
+
+      // update metadata for this post
+      // requires final content for hashing
+      if (metadata?.repository?.object?.__typename === 'Blob') {
+        const m = JSON.parse(
+          metadata.repository.object.text ?? '{}'
+        ) as MetadataSchema
+        m.generated = new Date().toISOString()
+        m.commit = hashFromUrl(metadata.repository.object.commitUrl)
+        ;(m.metadata ?? []).filter(
+          (c) =>
+            c.collection !== collection &&
+            (c.slug !== oldSlug || c.slug !== newSlug)
+        )
+        const state = MurmurHash3(content)
+        m.metadata.push({
+          ...matterData,
+          title: matterData.title,
+          publishedAt: matterData.publishedAt,
+          status: matterData.published,
+          slug: newSlug,
+          collection,
+          __outstatic: {
+            hash: `${state.result()}`,
+            commit: m.commit,
+            path: fullContentPath
```

Comment:
And at this level we shouldn't pass the `monoRepo` info.

```suggestion
            path: `${contentPath}/${collection}/${newSlug}.md`
```